darwin/dataset/utils.py

```diff
@@ -505,55 +505,57 @@ def _validate_inputs(
-        split_type (str, None): Type of dataset split. Can be 'random', 'stratified' or None.
+        partition (str, None): Dataset partition. Should be 'train', 'val', 'test', or None.
+        split_type (str, None): Type of dataset split. Can be 'random' or 'stratified'.
         annotation_type (str): Type of annotations. Can be 'tag', 'polygon', or 'bounding_box'.
 
     Raises:
         ValueError: If the input parameters do not match the expected values.
     """
     if partition not in ["train", "val", "test", None]:
-        raise ValueError("partition should be either 'train', 'val', 'test', or None")
-    if split_type not in ["random", "stratified", None]:
-        raise ValueError("split_type should be either 'random', 'stratified', or None")
+        raise ValueError("partition should be either 'train', 'val', 'test', or 'None'")
+    if split_type not in ["random", "stratified"]:
+        raise ValueError("split_type should be either 'random', or 'stratified'")
     if annotation_type not in ["tag", "polygon", "bounding_box"]:
         raise ValueError(
             "annotation_type should be either 'tag', 'bounding_box', or 'polygon'"
         )
 
 
-def _get_stems_from_split(
+def _get_annotation_filepaths_from_split(
     release_path: Path,
-    split: str,
-    split_type: Union[str, None],
     annotation_type: str,
-    partition: Union[str, None],
-) -> Generator:
+    partition: str,
+    split_type: str,
+    split: Optional[str] = "default",
+) -> Generator[str, None, None]:
     """
-    Determines the file stems based on the dataset split and other parameters.
+    Determines the filpaths based on the dataset split and other parameters.
 
     Args:
-        release_path (Path): Path to the dataset release.
-        split (str): Dataset split identifier.
-        split_type (str, None): Type of dataset split. Can be 'random', 'stratified' or None.
-        annotation_type (str): Type of annotations. Can be 'tag', 'polygon', or 'bounding_box'.
-        partition (str, None): Dataset partition. Should be 'train', 'val', 'test' or None.
+        release_path : Path
+            Path to the dataset release.
+        annotation_type : str
+            Type of annotations. Can be 'tag', 'polygon', or 'bounding_box'.
+        partition : str
+            Dataset partition. Should be 'train', 'val', 'test'.
+        split_type : str
+            Type of dataset split. Can be 'random' or 'stratified'.
+        split : Optional[str]
+            Dataset split identifier.
 
     Returns:
-        Generator[str]: File stems for the dataset.
+        Generator: [str, None, None]
+            Filepaths for the dataset.
 
     Raises:
         ValueError: If the split_type is invalid.
         FileNotFoundError: If the dataset partition file is not found.
     """
-    if split_type is None:
-        split_file = f"{partition}.txt"
-    elif split_type == "random":
+    if split_type == "random":
```

Comment:
should we add a default if we don't handle None?/why was this there before?

Reply:
Before the code gets to this stage, it passes the `split_type` into `_validate_inputs()` above. This function used to accept `split_type` as `None`, but since the top-level function `get_annotations()` sets `split_type` to `random` in the definition, it will never be `None`